File: app/src/main/java/oop2/ex02.java
package oop2;

public class ex02 {
    static class Animal {
        void sound() {
            System.out.println("...");
        }
    }

    static class Dog extends Animal {
        @Override
        void sound() {
            System.out.println("woof");
        }
    }

    static class Cat extends Animal {
        @Override
        void sound() {
            System.out.println("meow");
        }
    }

    public static void main(String[] args) {
        Animal[] animals = {new Dog(), new Cat()};
        for (Animal animal : animals) {
            animal.sound();
        }
    }
}

File: app-kotlin/src/main/kotlin/oop2/ex02.kt
package oop2

class ex02 {
    open class Animal {
        open fun sound() {
            println("...")
        }
    }

    class Dog : Animal() {
        override fun sound() {
            println("woof")
        }
    }

    class Cat : Animal() {
        override fun sound() {
            println("meow")
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val animals = arrayOf(Dog(), Cat())
            animals.forEach { it.sound() }
        }
    }
}
